libcasr/src/stacktrace.rs

```diff
@@ -287,11 +288,226 @@ impl Filter for Stacktrace {
             self.drain(pos + 1..);
         }
 
-        // Remove trusted functions from stack trace
+        // Find repeating intervals in stacktrace
+        let mut indices = Vec::new();
+        indices.resize(self.len(), true);
+        // We need to create an element that will not be equal to others
+        let mut not_an_element = StacktraceEntry::default();
+        not_an_element.debug.file = "not_equal_to_other_names".to_string();
+        not_an_element.debug.line = 1;
+        if let Ok(intervals) = get_interval_repetitions(self, &not_an_element) {
+            intervals.iter().for_each(|(start, end, seq_len)| {
+                (start + seq_len..end + 1).for_each(|idx| indices[idx] = false)
```

Comment:
We may just write this logic in filter

Reply:
Just advance intervals iterator when current index is above right border

Reply:
Thus, there is no need for `indices` vector

Reply:
In this case, as far as I understand, we will have +2 `if` branches in the filter loop: one check that the iterator has not ended, and another check for the boundary condition. Is it relevant instead of `Vec<bool>`?

Reply:
However, we won't use any extra memory

Reply:
done